cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -240,24 +241,35 @@ public String commitAndPush(final CommitMeta commitMeta,
           "No changes have been made for the applications to push.");
     }
 
-    git.add().addFilepattern(fileChanged.toString()).call();
+    for (Path fileChanged : filesChanged) {
+      git.add().addFilepattern(fileChanged.toString()).call();
+    }
 
     RevCommit commit = getCommitCommand(commitMeta).call();
 
-    String fileHash;
-    try {
-      fileHash = getFileHash(fileChanged, commit);
-    } catch (IOException e) {
-      throw new GitOperationException(
-          String.format("Failed to get fileHash for %s", fileChanged),
-          e);
+    Map<Path, String> fileHashes = new HashMap<>();
+    for (Path fileChanged : filesChanged) {
+      try {
+        String fileHash = getFileHash(fileChanged, commit);
```

Comment:
Seems like should have a batch `getFileHashes` method to do it more efficiently by doing a single TreeWalk instead of scanning it multiple times.

Reply:
updated. Added a new method getFileHashes to get the hashes using a single TreeWalk.